pkg/server/controllers.go

```diff
@@ -369,20 +367,8 @@ func (s *Server) installWorkloadResourceScheduler(ctx context.Context, config *r
 }
 
 func (s *Server) installWorkspaceScheduler(ctx context.Context, config *rest.Config) error {
-	controllerName := "kcp-workspace-scheduler"
-
-	// TODO(ncdc): this is here because the call to bootstrap.NewController below needs a bootstrap client for kcp,
-	// but it takes in a kcpclient.Interface, not kcpclient.ClusterInterface. The types on Server are all
-	// *.ClusterInterface. We'll be able to unify things once the work to simplify and consolidate our clients is
-	// done.
-	bootstrapConfig := rest.CopyConfig(config)
-	bootstrapConfig = rest.AddUserAgent(kcpclienthelper.SetMultiClusterRoundTripper(bootstrapConfig), controllerName)
-	bootstrapConfig.Impersonate.UserName = kcpBootstrapperUserName
-	bootstrapConfig.Impersonate.Groups = []string{bootstrappolicy.SystemKcpWorkspaceBootstrapper}
-
 	config = rest.CopyConfig(config)
```

Comment:
Global comment for this function - I feel like it would be safer (?) to copy the unmodified `config` param every time we create a new config; i.e., don't reassign via `config = ...`. WDYT? cc @stevekuznetsov

Reply:
I can see that.

Is there something that makes the copying less safe than in `installAPIBindingController`, or does the same comment apply there/any function that creates multiple controllers?

Reply:
Took me a few minutes to add the rest of the comments 😄